src/Entity/OgRole.php

```diff
@@ -177,8 +177,39 @@ public function setRoleType($role_type) {
   /**
    * {@inheritdoc}
    */
+  public function getName() {
+    // If the name is not set yet, try to derive it from the ID.
+    if (empty($this->name) && !empty($this->id())) {
+      list(, , $name) = explode('-', $this->id());
+      $this->setName($name);
+    }
+    return $this->name;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setName($name) {
+    $this->name = $name;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function save() {
-    if ($this->isNew()) {
+    // The ID of a new OgRole has to consist of the entity type ID, bundle ID
```

Comment:
Wouldn't it also be good to be able to have more generic roles? rather than always tied to a group? I am currently trying to use something similar. I have an admin type role, and this is available for on on all groups I create.